main.cpp

```diff
@@ -41,22 +41,458 @@ If you need inspiration for what to write, take a look at previously approved st
+        {
+            ledGainIndicator++;
+            cout << "ledGainIndicator blinks and increments" << endl;
+        }
+        else
+        {
+            ledGainIndicator--;
+            cout << "ledGainIndicator blinks and decrements" << endl;
+        }
+    }
+
+    cout << "ledGainIndicator is " << ledGainIndicator << endl;
+
+    return (ledGainIndicator);
+}
+
+Synth::Synth() : frequency1(400), frequency2(600), gain(0), attack(0.1f), release(1.0f)
+{
+    cout << "construct Synth" << endl;
+}
+Synth::~Synth()
+{
+    cout << "destruct Synth" << endl;
+}
+
+int Synth::modulateBeatFrequency( int newFreq1, int newFreq2 )
+{
+    frequencyOutput = std::abs(newFreq1 - newFreq2);
+    cout << "frequencyOutput is " << frequencyOutput << endl;
+    return frequencyOutput;
+
+}
+float Synth::changeAttack(float newAttack)
+{
+    attack = newAttack;
+    cout << "attack is " << attack << endl;
+    return attack;
+}
+float Synth::changeRelease(float newRelease)
+{
+    release = newRelease;
+    cout << "release is " << release << endl;
+    return release;
+}
+
+void Synth::setGain(int newLevel)
+{
+    uiControls.buttonPress(true);
+    uiControls.setKnob(newLevel);
+    uiControls.updateLEDIndicator();
+}           
 
 /*
  UDT 2:
  */
+ struct PastaShop
+ {
+    struct Style
+    {
+        Style() : 
+        whitePasta(false),
+        greenPasta(false),
+        flour(1.0),
+        thicknessOfDough(1.0)
+        {
+            cout << "constructing Style" << endl;
+        }
+
+        ~Style()
+        {
+            cout << "destructing Style" << endl;
+        }
+
+        void pastaColor(bool color1, bool color2);
+        string customShape(string s);
+        double addFlour(double doughWidth);
+
+        bool whitePasta;
+        bool greenPasta;
+        string name;
+        double flour;
+        double thicknessOfDough;
+    };
+
+    Style style;
+
+    PastaShop();
+    ~PastaShop();
+
+    double makeCustomPasta(double requestedAmount, string nameOfShape, bool color1, bool color2, double doughWidth);
+    double restockDough();
+    void pastaProfitTotal();
+
+    double amountOfDoughlBS;
+    double pastaProfit;
+    double pastaPrice {8.0};
+    double customRequestTotal {0.0};
+    double totalPastaWeight {0.0};
+
+ };
+
+void PastaShop::Style::pastaColor(bool color1, bool color2)
+{
+    if (color1 && color2)
+        cout << "white & green swirl" << endl;
+    else if(color1)
+        cout << "white pasta" << endl;
+    else if(color2)
+        cout << "green pasta" << endl;
+}
+
+string PastaShop::Style::customShape(string s)
+{
+    name = s;
+    return name;
+}
+
+double PastaShop::Style::addFlour(double doughWidth)
+{
+    if(doughWidth < 2.0 && doughWidth > 1.0)
+    {
+        cout << "add 1 cup flour" << endl;
+        flour = flour + 1.0;
+    }
+    else if(doughWidth > 2.0)
+    {
+        cout << "use a half cup less" << endl;
+        flour = flour - 0.5;
+    }
+    return flour;
```

Comment:
insert a blank line between your closing curly brace and the return statement to give the brain a visual break indicating the end of one coding idea and the start of a new one.
- [ ] revise